homeassistant/components/nest/__init__.py

```diff
@@ -134,6 +192,22 @@ async def async_handle_event(self, event_message: EventMessage) -> None:
             self._hass.bus.async_fire(NEST_EVENT, message)
 
 
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    if DATA_SDM not in config_entry.data:
+        # Legacy API
+        return True
+
+    if config_entry.version == 1:
+        _LOGGER.debug("Migrating from version %s", config_entry.version)
+        new: Dict[str, Any] = {**config_entry.data}
+        new["auth_implementation"] = f"{DOMAIN}.web"
```

Comment:
Can we avoid a hard migration and just check the entry data during setup and update it if needed. Doing a hard migration makes rolling back impossible and is something we want to avoid.

Reply:
Yes, no problem, didn't consider that. 

Sounds like it might be appropriate to use after a few stable releases to simplify code, but also this is not a super complex change to manage, so it could also never happen and be ok.